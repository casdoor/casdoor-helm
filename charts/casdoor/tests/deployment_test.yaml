suite: test deployment
templates:
- deployment.yaml

tests:
- it: should render with default values (sqlite)
  set:
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should set replicaCount (sqlite)
  set:
    replicaCount: 3
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should set custom image and pull secrets (sqlite)
  set:
    image:
      repository: my-repo
      name: my-casdoor
      tag: "1.0.0"
    imagePullSecrets:
    - name: my-secret
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should enable persistence (sqlite)
  set:
    persistence.enabled: true
    database.driver: sqlite
  asserts:
  - matchSnapshot: {}

- it: should add extra volumes and mounts (sqlite)
  set:
    extraVolumes:
    - name: extra-vol
      emptyDir: {}
    extraVolumeMounts:
    - name: extra-vol
      mountPath: /extra
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should disable probes (sqlite)
  set:
    probe:
      liveness:
        enabled: false
      readiness:
        enabled: false
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should enable ldap (sqlite)
  set:
    ldap.enabled: true
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should use config from secret (sqlite)
  set:
    configFromSecret: my-casdoor-secret
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should not have replicas when autoscaling is enabled (sqlite)
  set:
    autoscaling.enabled: true
    database.driver: sqlite
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should render with default values (postgres)
  set:
    database.driver: postgres
    persistence.enabled: false
  asserts:
  - matchSnapshot: {}

- it: should set replicaCount (postgres)
  set:
    replicaCount: 3
    database.driver: postgres
    persistence.enabled: false
  asserts:
  - matchSnapshot: {}
