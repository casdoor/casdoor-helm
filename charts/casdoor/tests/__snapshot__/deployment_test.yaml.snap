should add extra volumes and mounts (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
                - mountPath: /extra
                  name: extra-vol
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
            - emptyDir: {}
              name: extra-vol
should disable probes (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should enable ldap (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
                - containerPort: 10389
                  name: ldap
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should enable persistence (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should not have replicas when autoscaling is enabled (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should render with default values (postgres):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 2fa39bf0719a2f02822099856084a7eadd7e0d8455af37432266e7c0d2c04892
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
should render with default values (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should set custom image and pull secrets (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: my-repo/my-casdoor:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          imagePullSecrets:
            - name: my-secret
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should set replicaCount (postgres):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 3
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 2fa39bf0719a2f02822099856084a7eadd7e0d8455af37432266e7c0d2c04892
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
should set replicaCount (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 3
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-config
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
should use config from secret (sqlite):
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor
      template:
        metadata:
          annotations:
            checksum/config: 57b0042bfacc75c1183fd953e239e821fdb863dd570ba84cc554cf5a7e854a2e
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:v1.976.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /data
                  name: data-volume
          securityContext:
            fsGroup: 1000
            fsGroupChangePolicy: Always
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME-casdoor
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - secret:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: my-casdoor-secret
            - name: data-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-casdoor-data
